spa/src/pages/Connections/GHEnterpriseConnections/GHEnterpriseApplication.tsx

```diff
@@ -29,36 +42,92 @@ const applicationContentStyle = css`
 
 type GitHubEnterpriseApplicationProps = {
 	application: GitHubEnterpriseApplication;
+	setDataForModal: (
+		dataForModal: SuccessfulConnection | GitHubEnterpriseApplication
+	) => void;
+	setSelectedModal: (selectedModal: BackfillPageModalTypes) => void;
+	setIsModalOpened: (isModalOpen: boolean) => void;
 };
 
+function openChildWindow(url: string) {
+	const child: Window | null = window.open(url);
+	const interval = setInterval(function () {
+		if (child?.closed) {
+			clearInterval(interval);
+			AP.navigator.reload();
+		}
+	}, 100);
+	return child;
+}
+
 const GitHubEnterpriseApp = ({
 	application,
+	setIsModalOpened,
+	setDataForModal,
+	setSelectedModal,
 }: GitHubEnterpriseApplicationProps) => {
 	const [showAppContent, setShowAppContent] = useState<boolean>(true);
+	const toggleShowAppContent = () =>
+		setShowAppContent((prevState) => !prevState);
+	const onConnectNewApp = () => {
```

Comment:
I think this is an async method, cause `AP.context.getToken` is async.
So might need to add a loader or disable for the button when this method is called, otherwise the user can spam clicks and this could open multiple windows.